### PR TITLE
Remove fd from pollset when releasing the fd

### PR DESCRIPTION
**Pull request by yang-g**
This is caught in a test in another PR.

### CODE REVIEW

**Comment by yang-g** [comment]
ping @guantaol

File: src/core/lib/iomgr/ev_epoll1_linux.cc [hunk 1]
```diff
@@ -383,6 +383,10 @@ static void fd_shutdown_internal(grpc_fd* fd, grpc_error* why,
   if (fd->read_closure->SetShutdown(GRPC_ERROR_REF(why))) {
     if (!releasing_fd) {
       shutdown(fd->fd, SHUT_RDWR);
+    } else {
+      if (epoll_ctl(g_epoll_set.epfd, EPOLL_CTL_DEL, fd->fd, nullptr) != 0) {
```
Comment (guantaol) [inline src/core/lib/iomgr/ev_epoll1_linux.cc]: According to the Linux manual page http://man7.org/linux/man-pages/man2/epoll_ctl.2.html, the _event_ argument cannot be null for **EPOLL_CTL_DEL** in kernel versions before 2.6.9. Do we need to be portable to Linux kernels before 2.6.9?

Also, does any existing application expect that the released fd is still within the epoll set?

Comment (yang-g) [reply src/core/lib/iomgr/ev_epoll1_linux.cc]: Done. Global presubmit tests look fine too. PTAL

File: src/core/lib/iomgr/ev_epoll1_linux.cc [hunk 2]
```diff
@@ -383,6 +383,14 @@ static void fd_shutdown_internal(grpc_fd* fd, grpc_error* why,
   if (fd->read_closure->SetShutdown(GRPC_ERROR_REF(why))) {
     if (!releasing_fd) {
       shutdown(fd->fd, SHUT_RDWR);
+    } else {
+      /* we need a dummy event for earlier linux versions. */
+      epoll_event dummy_event;
+      memset(&dummy_event, 0, sizeof(dummy_event));
```
Comment (guantaol) [inline src/core/lib/iomgr/ev_epoll1_linux.cc]: Do we need to this memset? It looks redundant.

Comment (yang-g) [reply src/core/lib/iomgr/ev_epoll1_linux.cc]: Not sure what exactly is meant by "ignored" in the doc and just to be safe. If you prefer, I can remove it and re-run some tests.

Comment (guantaol) [reply src/core/lib/iomgr/ev_epoll1_linux.cc]: I prefer to remove it.

Comment (yang-g) [reply src/core/lib/iomgr/ev_epoll1_linux.cc]: Cool. Done.

**Pull request merged**